packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -25,6 +25,11 @@ final String _kZeroWidthSpace = new String.fromCharCode(0x200B);
 /// Used by [RenderEditable.onSelectionChanged].
 typedef void SelectionChangedHandler(TextSelection selection, RenderEditable renderObject, bool longPress);
 
+/// Signature for the callback that reports when the caret location changes.
+///
+/// Used by [RenderEditable.onCaretChanged].
+typedef void CaretChangedHandler(Rect caretRect);
```

Comment:
FWIW, `ValueChanged<Rect>` is the same signature. (It's fine to keep this though. We often have redundant typedefs, especially in cases where we can then hook more documentation to them.)